zerver/models.py

```diff
@@ -2139,6 +2164,12 @@ def has_permission(self, policy_name: str) -> bool:
         ]:
             raise AssertionError("Invalid policy")
 
+        if policy_name in Realm.REALM_PERMISSION_GROUP_SETTING_ID_TO_GROUP_SETTING:
+            allowed_user_group = getattr(
+                self.realm, Realm.REALM_PERMISSION_GROUP_SETTING_ID_TO_GROUP_SETTING[policy_name]
+            )
```

Comment:
We can have a separate function for group-based settings but I am not sure. We can wait for more feedback.

Reply:
But this may be fine since we would convert other settings to group based settings eventually.